### PR TITLE
CAS-611: Add Vote Type to Create Proposal

### PR DESCRIPTION
**Pull request by jbluks**
Adds a new section to the create proposal page for "Vote Type" which will currently be either Single Choice or Rank Voting

Screen Shot 2022-10-13 at 10 38 03 AM

### CODE REVIEW

File: frontend/packages/client/src/components/ProposalCreate/StepOne/index.js [hunk 1]
```diff
@@ -52,8 +55,14 @@ const StepOne = ({
...
       defaultValues: fieldsObj,
       resolver: yupResolver(stepOne.Schema),
+      context: { voteType },
     });
 
+  const handleVoteType = (voteType) => {
+    setVoteType(voteType);
+    setValue('voteType', voteType);
+  };
+
```
Comment (germanurrus) [inline frontend/packages/client/src/components/ProposalCreate/StepOne/index.js:65]: I think you can remove the state for holding voteType and passing it as a context to the hook by doing:
`const voteType = useWatch({ control, name: 'voteType' });`
then we just have the value in the form. (Below line 77)

File: frontend/packages/client/src/components/ProposalCreate/FormConfig.js [hunk 2]
```diff
@@ -33,14 +40,18 @@ const StepOneSchema = yup.object().shape({
         })
       ),
     })
-    .min(2, 'Please add a choice, minimum amout is two')
+    .when('$voteType', (voteType, schema) => {
+      return voteType === 'single-choice'
+        ? schema.min(2, 'Please add a choice, minimum amount is two')
+        : schema.min(3, 'Please add a choice, minimum amount is three');
+    })
```
Comment (germanurrus) [inline frontend/packages/client/src/components/ProposalCreate/FormConfig.js:47]: if you remove the context: { voteType } you can do: 
```
.when('voteType', (voteType, schema) =>
      voteType === 'single-choice'
        ? schema.min(2, 'Please add a choice, minimum amount is two')
        : schema.min(3, 'Please add a choice, minimum amount is three')
    )
```

File: frontend/packages/client/src/components/ProposalCreate/StepOne/index.js [hunk 3]
```diff
@@ -127,38 +136,195 @@ const StepOne = ({
...
+                </div>
+              </div>
+              <div className="py-5 pr-5">
+                <h5 className="title is-6 mb-2">Single Choice Voting</h5>
+                <p>
+                  Voters can only vote on one option and all options are
+                  customized by proposal creator.
+                </p>
+              </div>
+              <div className="has-background-light-grey p-4 rounded-sm-br rounded-sm-tr is-flex is-flex-direction-column is-align-self-stretch is-justify-content-center ">
```
Comment (germanurrus) [inline frontend/packages/client/src/components/ProposalCreate/StepOne/index.js:199]:
```suggestion
              <div className="has-background-light-grey p-4 px-2-mobile rounded-sm-br rounded-sm-tr is-flex is-flex-direction-column is-align-self-stretch is-justify-content-center">
```
Screen Shot 2022-10-14 at 10 43 44

File: frontend/packages/client/src/components/ProposalCreate/StepOne/index.js [hunk 4]
```diff
@@ -127,38 +136,195 @@ const StepOne = ({
...
+                  style={{ whiteSpace: 'nowrap' }}
+                >
+                  <div
+                    className="rounded-full has-background-grey has-text-white mr-2 is-flex is-align-items-center is-justify-content-center"
+                    style={{ width: 12, height: 12 }}
+                  ></div>
+                  <span className="smaller-text">Option C</span>
+                </div>
+              </div>
+            </div>
```
Comment (germanurrus) [inline frontend/packages/client/src/components/ProposalCreate/StepOne/index.js:233]: could you make this a static component so we can reduce  code here soma as for the other option

**Review by germanurrus — CHANGES_REQUESTED** [review]
Looks good! I wasn't able to create a proposal though, not sure what would be the issue. I was getting 
```
Error: {"message":"Error","errorCode":"ERR_1001","details":"There was an error trying to complete your request"}
    at checkResponse (utils.js:47:1)
    at async createProposalApiReq (proposals.js:66:1)

**Comment by jbluks** [comment]
> Looks good! I wasn't able to create a proposal though, not sure what would be the issue. I was getting
> 
> ```
> Error: {"message":"Error","errorCode":"ERR_1001","details":"There was an error trying to complete your request"}
>     at checkResponse (utils.js:47:1)
>     at async createProposalApiReq (proposals.js:66:1)
> ```

My initial guess is that the migration wasn't run - there is a new column in the proposals table.

File: frontend/packages/client/src/hooks/useProposalCreateMutation.js [hunk 5]
```diff
@@ -28,7 +28,7 @@ export default function useProposalCreateMuation() {
       if (!compositeSignatures && !voucher) {
         throw new Error('No valid user signature found.');
       }
-
+      console.log(proposalPayload);
```
Comment (germanurrus) [inline frontend/packages/client/src/hooks/useProposalCreateMutation.js:31]: output to be removed

**Review by germanurrus — APPROVED** [review]
LGTM!(FE part)

**Pull request merged**